### ISSUE TITLE
Widgets editor: editing block deletes its content when widgets editor is disabled in WP 5.7

### ISSUE THREAD

**Issue by hellofromtonya** [WordPress/gutenberg#31890]
## Description
When the widgets editor is disabled, editing a widget block causes its content to be deleted. When re-enabling the widgets editor, the blocks and widgets no longer appear in the Widgets UI and an error appears in the console.

## Step-by-step reproduction instructions
1. Deactivate Classic Widgets plugin
2. Go to Appearance → Widgets
3. Add a block to a widget area
4. Update
5. Go to Plugins → Installed Plugins
6. Activate the Classic Widgets plugin
7. Go to Appearance → Widgets
8. Edit the content in a widget block
9. Click "Save" button => _Notice the content in the "Block HTML" field is deleted_
10. Go to Plugins → Installed Plugins
11. Deactivate Classic Widgets plugin
12. Go to Appearance → Widgets
13. Expand the widget area and inactive widgets => _Notice no widgets or blocks display_
14. Add a new block => _Notice the "Update" button is inactive_

To recover:

1. Go to Plugins → Installed Plugins
2. Activate Classic Widgets plugin
3. Go to Appearance → Widgets
4. Delete all "Block" widgets
5. Go to Plugins → Installed Plugins
6. Deactivate Classic Widgets plugin
7. Go to Appearance → Widgets => Notice the widgets reappear and no console error

## Expected behaviour
- Able to edit the block's HTML content
- No console error when reactivating the feature

## Actual behaviour
Editing a block when the widgets editor is disable causes:

- Deletes the block's content
- Disables loading all blocks and widgets in the Widgets Editor when the feature is re-enabled
- Disables ability to update changes
- Error in console
```
index.js?ver=99a792e47c3c4aae684bc25061ba6c1b:1 Uncaught (in promise) TypeError: Cannot read property 'length' of null
    at a (index.js?ver=99a792e47c3c4aae684bc25061ba6c1b:1)
    at p (index.js?ver=99a792e47c3c4aae684bc25061ba6c1b:1)
    at f (index.js?ver=99a792e47c3c4aae684bc25061ba6c1b:1)
    at kr (index.js?ver=8f1cce1c350691cf7d5c64dd71fc85bd:13)
    at I (index.js?ver=b98aa486c518ea11a026635bfec431bc:15)
    at Object.G [as fulfill] (index.js?ver=b98aa486c518ea11a026635bfec431bc:19)
    at G.next (<anonymous>)
    at index.js?ver=f43d3e5edac40e5bae37689f61e35ab0:1
    at index.js?ver=f43d3e5edac40e5bae37689f61e35ab0:1
    at index.js?ver=f43d3e5edac40e5bae37689f61e35ab0:1
```

## Screenshots or screen recording (optional)

![edit-widget-block-removes-widgets](https://user-images.githubusercontent.com/7284611/118500425-643a9f80-b6ed-11eb-9f37-ebe2ae37d33f.gif)

## WordPress information
- WordPress version: 5.7.2
- Gutenberg version: 10.6.0
- Are all plugins except Gutenberg deactivated? Yes, except for when using the Classic Widgets plugin as noted above
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes, Twenty Twenty-One

## Device information
- Device: Desktop
- Operating system: MacOS Big Sur
- Browser: Chrome 90.0, Firefox 88.0.1, Safari 14.1

**Comment by noisysocks** [comment]
I suspect this is because `WP_Widget` was only registered when the theme opted into widgets and that this is now fixed in Core. Haven't verified this, yet, though 🙂

**Comment by noisysocks** [comment]
Yes, this is only a problem in the plugin. I'll de-prioritise this bug.

**Comment by bobbingwide** [comment]
Interesting, we got a very similar message when demoing the Block widget editor in wp-pompey.org.uk on 21st July. 
WordPress 5.8, with a child theme of Twenty Seventeen.

```
Uncaught (in promise) TypeError: Cannot read property 'length' of null
    at a (block-serialization-default-parser.min.js:2)
    at p (block-serialization-default-parser.min.js:2)
    at f (block-serialization-default-parser.min.js:2)
    at Ar (blocks.min.js:2)
    at T (edit-widgets.min.js:2)
    at Object.q [as fulfill] (edit-widgets.min.js:2)
    at q.next (<anonymous>)
    at redux-routine.min.js:2
    at redux-routine.min.js:2
    at redux-routine.min.js:2
```

It went wrong during the demo. It was working fine until I added the Meta Legacy Widget, which didn't work.

See https://www.youtube.com/watch?v=SpQGAgShYoY

The widgets stopped appearing in the editor at 6:40

https://youtu.be/SpQGAgShYoY?t=400


BTW: I'd not done anything with the Classic Widgets plugin. 
Is this the same issue?

**Comment by bobbingwide** [comment]
### Workaround for my problem

1. Install and activate the Classic Widget plugin. 
2. Delete the empty block
3. Deactivate the Classic Widget plugin


https://github.com/wppompey/wp-pompey/issues/6#issuecomment-891792319

**Comment by noisysocks** [comment]
Closing this as "fixed" because Gutenberg now requires WP 5.8 at minimum which means `WP_Widget_Block` will always exist.

@bobbingwide: I think you have a seperate problem there. Could you please re-test against the latest versions of WordPress and Gutenberg and open a new issue if you're still seeing this?
